apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts

```diff
@@ -0,0 +1,364 @@
+    // Default sequence (start at 1, triggered = false)
+    sequencer.newSequence();
+    expect(sequencer.getLastMeasure()).to.equal(1);
+
+    sequencer.startFunctionContext('when_run');
+    sequencer.playSound('id', 'blockId');
+    expect(sequencer.getPlaybackEvents()[0].triggered).to.be.false;
+
+    sequencer.clear();
+
+    // Triggered sequence
+    sequencer.newSequence(10, true);
+    expect(sequencer.getLastMeasure()).to.equal(10);
+
+    sequencer.startFunctionContext('when_run');
+    sequencer.playSound('id', 'blockId');
+    expect(sequencer.getPlaybackEvents()[0].triggered).to.be.true;
+  });
+
+  it('Adds a sound event', () => {
+    sequencer.newSequence();
+    sequencer.startFunctionContext('when_run');
+    sequencer.playSound('id', 'blockId');
+
+    Sinon.assert.calledWith(library.getSoundForId, 'id');
+
+    const playbackEvents = sequencer.getPlaybackEvents();
+    expect(playbackEvents.length).to.equal(1);
+
+    const event = playbackEvents[0];
+    expect(event.type).to.equal('sound');
+    expect(event.id).to.equal('id');
+    expect(event.blockId).to.equal('blockId');
+    expect(event.when).to.equal(1);
+    expect(event.length).to.equal(testSound.length);
+    expect(event.triggered).to.be.false;
+    expect((event as SoundEvent).soundType).to.equal(testSound.type);
+  });
+
+  it('Adds a pattern event', () => {
+    const patternValue: PatternEventValue = {
+      kit: 'machine',
+      events: [],
+    };
+
+    sequencer.newSequence();
+    sequencer.startFunctionContext('when_run');
+    sequencer.playPattern(patternValue, 'blockId');
+
+    const playbackEvents = sequencer.getPlaybackEvents();
+    expect(playbackEvents.length).to.equal(1);
+
+    const event = playbackEvents[0];
+    expect(event.type).to.equal('pattern');
+    expect(event.id).to.equal(JSON.stringify(patternValue));
+    expect(event.blockId).to.equal('blockId');
+    expect(event.when).to.equal(1);
+    expect(event.length).to.equal(DEFAULT_PATTERN_LENGTH);
+    expect(event.triggered).to.be.false;
+    expect((event as PatternEvent).value).to.equal(patternValue);
+  });
+
+  it('Adds a chord event', () => {
+    const chordValue: ChordEventValue = {
+      instrument: 'piano',
+      notes: [],
+      playStyle: 'arpeggio-up',
+    };
+
+    sequencer.newSequence();
+    sequencer.startFunctionContext('when_run');
+    sequencer.playChord(chordValue, 'blockId');
+
+    const playbackEvents = sequencer.getPlaybackEvents();
+    expect(playbackEvents.length).to.equal(1);
+
+    const event = playbackEvents[0];
+    expect(event.type).to.equal('chord');
+    expect(event.id).to.equal(JSON.stringify(chordValue));
+    expect(event.blockId).to.equal('blockId');
+    expect(event.when).to.equal(1);
+    expect(event.length).to.equal(DEFAULT_PATTERN_LENGTH);
+    expect(event.triggered).to.be.false;
+    expect((event as ChordEvent).value).to.equal(chordValue);
+  });
+
+  it('nests together and sequential events correctly', () => {
```

Comment:
ultra nit: consistency in upper-casing (or not) these names :)

Reply:
Done!